### ISSUE TITLE
import/no-unresolved false positive with locally installed npm package

### ISSUE THREAD

**Issue by sergioadimedia** [import-js/eslint-plugin-import#2604]
Hello guys!

I have a package installed from a file instead of installing it from npm's registry.
This is how the relevant part of the `package.json` looks:
```
"dependencies": {
    "exceljs": "file:pre-built-dependencies/exceljs"
}
```
However, I'm getting an `import/no-unresolved` when I use it.
Here's how I import the package:
`import ExcelJS from "exceljs";`

When I check the `node_modules` directory I can see that `exceljs` is a symbolic link which points to `pre-built-dependencies/exceljs`.

I have tried adding some configuration parameters to eslint but I have not got it to work. This is how the relevant part of `.eslintrc` looks like:
```
"settings": {
    "import/resolver": {
      "node": {
        "extensions": [".js", ".jsx"],
        "paths": ["src"],
        "moduleDirectory": ["node_modules", "pre-built-dependencies"]
      }
    }
  }
```

Needless to say that the project is working, so the package is correctly installed.

I don't know what I'm doing wrong or which config parameter I'm missing.
Thank you in advance!

**Comment by ljharb** [comment]
This is because we use resolve v1, which defaults the `preserveSymlinks` option to `true`, matching node versions at the time v1 was released. v2 will default it to `false`, matching modern node versions.

If you add `"preserveSymlinks": false` inside the `node` object in your config, does that fix it?

**Comment by sergioadimedia** [comment]
Any update on this topic?

**Comment by ljharb** [comment]
No update. What if you use a URL of `file:./pre-built-dependencies/exceljs`?
